ImperatorToCK3/Source/CK3Outputter/outTitles.cpp

```diff
@@ -1,20 +1,45 @@
 #include "outTitles.h"
 #include <filesystem>
 #include <fstream>
+#include "../commonItems/OSCompatibilityLayer.h"
+#include "../commonItems/CommonFunctions.h"
 
 
-void CK3::outputTitles(const std::string& outputModName, const std::map<std::string, std::shared_ptr<Title>>& titles)
+void CK3::outputTitles(const std::string& outputModName, const std::string& ck3Path, const std::map<std::string, std::shared_ptr<Title>>& titles)
 {
-	unsigned char bom[] = { 0xEF,0xBB,0xBF };
-
-	for (const auto& [first, second] : titles)
+	// blank all title history files from vanilla
+	auto fileNames = commonItems::GetAllFilesInFolderRecursive(ck3Path + "/game/history/titles/");
+	for (const auto& fileName : fileNames)
+	{
+		std::ofstream file("output/" + outputModName + "/history/titles/" + fileName);
+		file.close();
+	}
+	
+	for (const auto& [name, title] : titles)
 	{
-		std::ofstream output("output/" + outputModName + "/common/landed_titles/" + first + ".txt"); // dumping all into one file
-		if (!output.is_open())
+		if (!(name.find("c_")==0)) // title is not a county
```

Comment:
wait, you don't output counties? What if you have standalone OPM counties?

Reply:
At the moment, I don't. I'm not modifying de jure yet, so the only titles I need to output here ATM are generated `e_IMPTOCK` titles.